backend/app/controllers/MenteesController.php

```diff
@@ -75,4 +176,50 @@ public function addMentee()
             ]);
         }
     }
+
+    public function firstLogin()
+    {
+
+        $users_id = auth()->id();
+        $initial_weight = request()->get('initial_weight');
+        $current_weight = request()->get('current_weight');
+        $target_weight = request()->get('target_weight');
+        $growth = request()->get('growth');
+        $age = request()->get('age');
+        $activity_factor = request()->get('activity_factor');
+        $sex = request()->get('sex');
+
+        if(!$initial_weight || !$current_weight || !$target_weight || !$growth || !$age || !$activity_factor || !$sex) {
+            response()->exit([
+                'error' => '404',
+                'message' => "One of required fields isn't filled.",
+            ]);
+        }
+
```

Comment:
I think there must be a better way to handle validation. Right now we don't even know what type of data we receive.

Beside that it should be a `400`

Reply:
Yes, there is another way.
However, after talking to one of the authors of Leaf, one bug was found, which is currently unavoidable.
Using their built-in function https://leafphp.dev/modules/forms/v/1.2/#validation
There are bugs and they have opened a ticket on github.

Reply:
Good research 👍 we can leave it as it is now and consider a separate validation library soon (in another task/PR)